cpp/src/arrow/compute/api_vector.h

```diff
@@ -188,6 +188,35 @@ class ARROW_EXPORT PartitionNthOptions : public FunctionOptions {
   NullPlacement null_placement;
 };
 
+/// \brief Options for IsMonotonic
+class ARROW_EXPORT IsMonotonicOptions : public FunctionOptions {
+ public:
+  enum NullHandling {
+    /// Ignore nulls.
+    IGNORE_NULLS,
+    /// Use min value of element type as the value of nulls.
+    /// -Inf for floating point numbers.
+    USE_MIN_VALUE,
+    /// Use max value of element type as the value of nulls.
+    /// Inf for floating point numbers.
+    USE_MAX_VALUE
+  };
```

Comment:
[Ordering of nulls and NaNs were also discussed in the sorting function](https://github.com/apache/arrow/pull/11212/files#diff-d8e3089c085273e425a5106e77566f2e1348c483a79eeb544a4c9ee0e8327f34R89-R96). I would expect that `IsMonotonic` and sorting function are consistent. That is, if a sort operation is performed first, then the corresponding `IsMonotonic` should result in true.

Reply:
Yes, I agree that a sort before invoking this kernel should result in true for the corresponding check. However I feel the null handling variants are a bit confusing: `AtStart` defines `NaN > null` and `AtEnd` defines `NaN < null`. Also, the sorting kernel can ignore equality, but this kernels considers it to check if values are unique (strictly increasing/decreasing).

I think if we want to allow users to define order of unordered values (both for sorting and this kernel) we need something like this:
```c++
bool compare_nulls = false; // default: any null results in false outputs (or error in case of sort)
bool compare_nans = false; // default: any nan results in false outputs (or error in case of sort)

// these are not needed when sorting
bool nulls_equal = false; // when nulls are compared, are they considered equal?
bool nans_equal = false; // when nans are compared, are they considered equal?

// when both nulls and nans are compared
enum Ordering { Less, Equal, Greater }
Ordering nan_compared_with_null; // when comparing nulls and nans, what ordering should be used?
```